components/sup/src/event.rs

```diff
@@ -177,6 +179,26 @@ pub fn service_stopped(service: &Service) {
     }
 }
 
+/// Send an event at the start of a Service update.
+pub fn service_update_started(service: &Service, update: &PackageIdent) {
+    if stream_initialized() {
+        publish(ServiceUpdateStartedEvent { event_metadata:   None,
+                                            service_metadata: Some(service.to_service_metadata()),
+                                            update_origin:    update.origin.clone(),
+                                            update_name:      update.name.clone(),
+                                            update_version:
+                                                update.version
+                                                      .clone()
+                                                      .expect("Can't send service update started \
```

Comment:
This seems like an awkward way to panic. The ideal would be to have this function take a `FullyQualifiedPackageIdent` type, but until we have that, I'm not sure we want to crash the entire supervisor. Would something like this be preferable?
```rust
        if let PackageIdent { origin: update_origin,
                              name: update_name,
                              version: Some(update_version),
                              release: Some(update_release), } = update.clone()
        {
            publish(ServiceUpdateStartedEvent { event_metadata: None,
                                                service_metadata: service.into(),
                                                update_origin,
                                                update_name,
                                                update_version,
                                                update_release });
        } else {
            error!(…);
        }
    }
```

Reply:
@baumanj thanks, i believe this is a non-issue now that i've refactored the fully-qualified ident for the update package into a string in the proto (the same way the current package ident is expressed). i'll keep this in mind going forward, however!